generators/entity/index.js

```diff
@@ -994,9 +994,22 @@ class EntityGenerator extends BaseBlueprintGenerator {
                         }
                     }
 
+                    relationship.otherEntityFieldName = 'id';
                     if (otherEntityName === 'user') {
                         relationship.otherEntityTableName = `${jhiTablePrefix}_user`;
                         hasUserField = true;
+                    } else if (otherEntityName === 'authority') {
+                        relationship.otherEntityTableName = `${jhiTablePrefix}_authority`;
+                        relationship.otherEntityFieldName = 'name';
+                        if (relationship.otherEntityField === 'id') {
```

Comment:
I think otherEntityField is used for bi-directional relationships.
Is this needed?

Reply:
it is for the displayed field in the client. If you don't specify this one in the subgenerator or JDL, say

```
relationship ManyToOne {
 A{authority} to Authority
}
```

we assume 'id' to be the display field. This will lead to compile errors. So that's way this fall back.